core/templates/dev/head/services/ExplorationFeaturesService.ts

```diff
@@ -17,44 +17,52 @@
  *               the exploration editor.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-oppia.factory('ExplorationFeaturesService', [function() {
-  var settings = {
-    areParametersEnabled: false,
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationFeaturesService {
+  static settings = {
```

Comment:
is there any specific reason to make this static?

Can't we directly use `isImprovementsTabEnabled`, `isPlaythroughRecordingEnabled`, `areParametersEnabled`?

Reply:
This is static since the uses in the codebase are inferring that the class should be static, like this one: https://github.com/oppia/oppia/blob/109108cbfc65c20847a7481c4290c579740e808d/core/templates/dev/head/pages/exploration-editor-page/statistics-tab/statistics-tab.directive.ts#L122-L123

Reply:
Ah, I was asking to have this.isPlaythroughRecordingEnabled, this.isImprovementsTabEnabled .. instead of `ExplorationFeaturesService.settings.isPlaythroughRecordingEnabled` etc. like [this](https://angular.io/api/upgrade/static/downgradeInjectable#examples). (`ExplorationFeaturesService.settings.isPlaythroughRecordingEnabled` looks bit wierd to me!)

I realized that we follow `this` pattern for factory objects in angularJS. I'm okay with this!

Reply:
Ah

Reply:
Yeah actually we tried using `this` but there is some problem regarding class properties and stuff.